compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1771,6 +1790,43 @@ object Types {
       ctx.underlyingRecursions -= 1
     }
 
+    // def noArg = throw new AssertionError(s"$pre contains no matching argument for ${sym.showLocated}")
+
+    /** The argument corresponding to class type parameter `tparam` as seen from
+     *  prefix `pre`.
+     */
```

Comment:
Complication: Need specific logic to substitute an actual argument for a formal parameter. This is somewhat more involved than the previous matchParams, which was removed.